### PR TITLE
Resolve security vulnerabilities

### PR DESCRIPTION
**Pull request by marianan**
Resolved security vulnerabilities reported by npm audit (see details below)
Ran unit tests - 16 passing (552ms)

Details:
c:\Work\Git\vscode-azure-iot-edge>npm audit

                       === npm audit security report ===

# Run  npm install --save-dev mocha@8.1.3  to resolve 1 vulnerability
SEMVER WARNING: Recommended action is a potentially breaking change

  Low             Prototype Pollution

  Package         minimist

  Dependency of   mocha [dev]

  Path            mocha > mkdirp > minimist

  More info       https://npmjs.com/advisories/1179

# Run  npm update bl --depth 7  to resolve 3 vulnerabilities

  High            Remote Memory Exposure

  Package         bl

  Dependency of   download-git-repo

  Path            download-git-repo > download > decompress > decompress-tar >
                  tar-stream > bl

  More info       https://npmjs.com/advisories/1555

  High            Remote Memory Exposure

  Package         bl

  Dependency of   download-git-repo

  Path            download-git-repo > download > decompress >
                  decompress-tarbz2 > decompress-tar > tar-stream > bl

  More info       https://npmjs.com/advisories/1555

  High            Remote Memory Exposure

  Package         bl

  Dependency of   download-git-repo

  Path            download-git-repo > download > decompress > decompress-targz
                  > decompress-tar > tar-stream > bl

  More info       https://npmjs.com/advisories/1555

                                 Manual Review
             Some vulnerabilities require your attention to resolve

          Visit https://go.npm.me/audit-guide for additional guidance

  Low             Prototype Pollution

  Package         minimist

  Patched in      >=0.2.1 <1.0.0 || >=1.2.3

  Dependency of   vscode [dev]

  Path            vscode > mocha > mkdirp > minimist

  More info       https://npmjs.com/advisories/1179

found 5 vulnerabilities (2 low, 3 high) in 687 scanned packages
  run `npm audit fix` to fix 3 of them.
  1 vulnerability requires semver-major dependency updates.
  1 vulnerability requires manual review. See the full report for details.

c:\Work\Git\vscode-azure-iot-edge>npm audit

                       === npm audit security report ===

found 0 vulnerabilities
 in 756 scanned packages

c:\Work\Git\vscode-azure-iot-edge>

### CODE REVIEW

**Comment by SLdragon** [comment]
Best practice to fix the security issue:

1. Try to use ```npm audit fix``` to fix the issue

2. If some dependencies cannot be resolved by the ```npm audit fix```, then you can create issues to the dependency repo to remind them to fix it. When they fix it, then we can use ```npm audit fix``` to fix it.

3. If some dependencies cannot be resolved and they only used in the test framework, you can ignore it

4. Do not force change the version, due to it may introduce unexpected behavior if there are some breaking changes

File: package.json [hunk 1]
```diff
@@ -513,6 +513,7 @@
     ]
   },
   "scripts": {
+    "preinstall": "npx npm-force-resolutions",
```
[the remark this answers is not in the document]
Comment (marianan) [reply package.json:516]: the force fix will only apply to the minimist package, see resolutions below so it should be safe to use?

---
In reply to: [490672278](https://github.com/microsoft/vscode-azure-iot-edge/pull/560#discussion_r490672278) [](ancestors = 490672278)

**Comment by marianan** [comment]
Thank you for sharing these practices! Could you share how you disable test framework issues in GitHub, I will open tracking issues for the remaining dependencies.

---
In reply to: [694621445](https://github.com/microsoft/vscode-azure-iot-edge/pull/560#issuecomment-694621445) [](ancestors = 694621445)

**Comment by SLdragon** [comment]
> Thank you for sharing these practices! Could you share how you disable test framework issues in GitHub, I will open tracking issues for the remaining dependencies.
> 
> In reply to: [694621445](https://github.com/microsoft/vscode-azure-iot-edge/pull/560#issuecomment-694621445) [](ancestors = 694621445)

You can follow this picture to disable the security alert in the Github when this alert can be ignored:

image

**Review by SLdragon — APPROVED** [review]

**Pull request merged**